### PR TITLE
Add support to blacklist entities

### PR DESCRIPTION
**Pull request by heinwessels**
I implemented it as a regular interface. So another mod would have to depend on PipeVisualizer and expand the blacklist during `on_init` and `on_configuration_changed`. It is possible for mods to change the blacklist at random times I guess, but without adding extra complexity I don't see an easy way around that.

I also didn't write an explanation on how this interface should be used. Not sure if you want to do that yourself.

### CODE REVIEW

**Comment by heinwessels** [comment]
**Update:** After a chat with Wiwiweb changed it to use the reverse interface method. If you would like it reverted, just let me know. No dependencies are now required to make this work.

For completeness this is a minimal version of how I implemented the other side in FluidicPower
```lua
remote.add_interface("FluidicPower", {
    ["pipe_visualizer_blacklist"] = function()
        return {"fluidic-energy-sensor", "fluidic-transformer" }
    end,
})
```

**Comment by raiguard** [comment]
I'm not a huge fan of reverse interfaces, so I would appreciate it if you could revert that. Thanks for the contribution!

**Comment by heinwessels** [comment]
Okay, changed it back. Now it's reset on every config change, so there's no persistence. A mod that interacts with it will have to call it on `init`, and `on_config_changed`, and have an dependency on `PipeVisualizer`. For example:

```lua
local function support_pipe_visualizer()
    blacklist = { "fluidic-energy-sensor", "fluidic-transformer", ...}
    if remote.interfaces["PipeVisualizer"] and remote.interfaces["PipeVisualizer"]["blacklist"] then
        remote.call("PipeVisualizer", "blacklist", blacklist)
    end
end

script.on_init(support_pipe_visualizer)
script.on_configuration_changed(support_pipe_visualizer)
```

File: control.lua [hunk 1]
```diff
@@ -8,4 +8,5 @@ handler.add_libraries({
   require("__PipeVisualizer__/scripts/order"),
   require("__PipeVisualizer__/scripts/overlay"),
   require("__PipeVisualizer__/scripts/renderer"),
+  require("__PipeVisualizer__/scripts/blacklist"),
```
Comment (raiguard) [inline control.lua]: These should be alphabetized.

File: scripts/blacklist.lua [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    blacklist = function(entity_names)
+      if type(entity_names) ~= "table" then return end
+      for _, entity_name in pairs(entity_names) do
+        global.blacklist[entity_name] = true
+      end
+    end,
+  })
+end
+
+local function reset(e)
```
Comment (raiguard) [inline scripts/blacklist.lua:18]: Missing type annotation.

**Review by raiguard — CHANGES_REQUESTED** [review]
Thanks! Just a few things. Also, please install [stylua](https://github.com/JohnnyMorganz/StyLua) and run the code through it to enforce my style rules.

File: scripts/blacklist.lua (hunk 2 above)
Comment (heinwessels) [reply scripts/blacklist.lua:18]: I did the following, not sure if its correct.
```lua
---@param e ConfigurationChangedData?
local function reset(e)
```

**Comment by heinwessels** [comment]
Comments addressed, and ran it through Stylua :)

**Comment by raiguard** [comment]
Thanks!

**Pull request merged**